hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/insights/om/om.tsx

```diff
@@ -777,6 +820,66 @@ export class Om extends React.Component<Record<string, object>, IOmdbInsightsSta
     });
   };
 
+  // Pending Delete Directories
+  fetchDeletePendingDir = (limit: number, prevKeyDirDelete: string) => {
+    this.setState({
+      loading: true,
+      nextClickable: true,
+      prevClickable: true
+    });
+
+    //Cancel any previous pending request
+    cancelRequests([
+      cancelMismatchedEndpointSignal,
+      cancelOpenKeysSignal,
+      cancelDeletePendingSignal,
+      cancelDeletedKeysSignal,
+      cancelRowExpandSignal,
+      cancelDeletedPendingDirSignal
+    ]);
+
+    let deletePendingDirEndpoint;
+    if (prevKeyDirDelete === "" || prevKeyDirDelete === undefined ) {
+      deletePendingDirEndpoint = `/api/v1/keys/deletePending/dirs?limit=${limit}`;
+    }
+    else {
+      deletePendingDirEndpoint = `/api/v1/keys/deletePending/dirs?limit=${limit}&prevKey=${prevKeyDirDelete}`;
+    }
+
+    const { request, controller } = AxiosGetHelper(deletePendingDirEndpoint, cancelDeletedPendingDirSignal);
+    cancelDeletedPendingDirSignal = controller 
+    request.then(deletePendingDirResponse => {
+      const deletedDirInfo = deletePendingDirResponse && deletePendingDirResponse.data && deletePendingDirResponse.data.deletedDirInfo;
+      if (deletePendingDirResponse && deletePendingDirResponse.data && deletePendingDirResponse.data.lastKey === "") {
+        //last key of api is empty may be last record no further records
+        dirPendingPrevList =[""];
+        this.setState({
+          loading: false,
+          nextClickable: false,
+          pendingDeleteDirDataSource: deletedDirInfo
+        })
+      }
+      else {
+           if (this.state.prevKeyDirDelete === "" || this.state.prevKeyDirDelete === undefined ) {
+            this.setState({ prevClickable: false })
+        }
+        if (dirPendingPrevList.includes(deletePendingDirResponse.data.lastKey) === false) {
+          dirPendingPrevList.push(deletePendingDirResponse.data.lastKey);
+        }
```

Comment:
```suggestion
      if (!dirPendingPrevList.includes(lastKey)) {
        dirPendingPrevList.push(lastKey);
      }
```

Reply:
Yes Need to change logic  once this PR gets merged. once https://github.com/apache/ozone/pull/5658